### ISSUE TITLE
InstantSearch component updating status inside unrelated InstantSearch component 

### ISSUE THREAD

**Issue by johnschatner** [algolia/instantsearch#6116]
### 🐛 Current behavior

Consider this scenario:
```
<Header>
   <InstantSearch id="1" {...relevantProps}>
       <SearchBox />
   </InstantSearch>
</Header>

<SearchPage>
   <InstantSearch id="2" {...relevantProps}>
       <SearchBox style={{ display: "none" }} />
       <Hits />
   </InstantSearch>
</SearchPage>
```

The issue is that inside SearchPage I want the loading state of the second (id: 2) instantsearch component. I do this with the useInstantSearch hook to retrieve the status. The peculiar thing is that whenever I click the SearchBox in the header (id: 1), the status of the second InstantSearch component reacts to this and updates the status object, setting it to "loading", then "stalled", then "idle".

### 🔍 Steps to reproduce

Try above scenario.

### Live reproduction

n/a

### 💭 Expected behavior

I expect the status state to be seperate and not react to unrelated InstantSearch components.

### Package version

instantsearch.js@4.66.1, react-instantsearch@7.7.1

### Operating system

Windows 11

### Browser

Chrome 122.0.6261.131

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by dhayab** [comment]
This shouldn't happen, I've reproduced your scenario in this sandbox and it works as expected: https://codesandbox.io/p/sandbox/gh-6116-forked-24zxfn. Can you give more details about your implementation, and why you want to use 2 InstantSearch in the same page?

**Comment by johnschatner** [comment]
Thanks for your sandbox dhayab. I found the issue. I was managing a state inside the `<SearchContext />` called `searchIsActive`. The context also holds the searchClient. I wrapped the entire app with this SearchContext so any <InstantSearch /> component reacted to the state updates to `searchIsActive`.

Here's a reproduction sandbox: [](https://codesandbox.io/p/sandbox/gh-6116-forked-d328gf)

The problems lies in the following files:
* SearchContext (managing searchIsActive state)
* CustomSearchBox (useSearch hook and setting the searchIsActive state on events)

So when I interact with the header search box the status updates in the search page. The reason for wanting this state is for controlling a modal with a list of hits under the search box.

The workaround was to not have this state in the context but instead in the nearest common parent for any component that needs the state and prop drill it.
